src/providers/BasicPoolsProvider.tsx

```diff
@@ -0,0 +1,208 @@
+  poolName: string
+  targetAddress: string | null
+  tokens: string[]
+  underlyingTokens: string[] | null
+  basePoolAddress: string | null
+  metaSwapDepositAddress: string | null
+  isSaddleApproved: boolean
+  isRemoved: boolean
+  isGuarded: boolean
+  isPaused: boolean
+  virtualPrice: BigNumber
+  adminFee: BigNumber
+  swapFee: BigNumber
+  aParameter: BigNumber
+  tokenBalances: BigNumber[]
+  underlyingTokenBalances: BigNumber[] | null
+  lpTokenSupply: BigNumber
+  miniChefRewardsPid: number | null
+  isMetaSwap: boolean
+}
+
+type BasicPool = {
+  isMigrated: boolean
+  sdlPerDay: BigNumber
+} & SwapInfo
+type BasicPools = Record<string, BasicPool> | null
+
+export const BasicPoolsContext = React.createContext<BasicPools>(null)
+
+export default function BasicPoolsProvider({
+  children,
+}: React.PropsWithChildren<unknown>): ReactElement {
+  const { chainId, library } = useActiveWeb3React()
+  const [basicPools, setBasicPools] = useState<BasicPools>(null)
+  const { lastTransactionTimes } = useSelector(
+    (state: AppState) => state.application,
+  )
+
+  useEffect(() => {
+    async function fetchBasicPools() {
+      if (!chainId || !library) {
+        setBasicPools(null)
+        return
+      }
+      const pools = await getPoolsBaseData(library, chainId)
+      const poolsAddresses = pools.map((pool) => pool.poolAddress)
+      const migrationData = await getMigrationData(
+        library,
+        chainId,
+        poolsAddresses,
+      )
+      const result = pools.reduce((acc, pool) => {
+        const poolData = { ...pool } as BasicPool
+        poolData.isMigrated = migrationData?.[pool.poolAddress] != null
+        return {
+          ...acc,
+          [pool.poolAddress]: poolData,
+        }
+      }, {} as BasicPools)
+      setBasicPools(result)
+    }
+    void fetchBasicPools()
+  }, [chainId, library, lastTransactionTimes])
+
+  return (
+    <BasicPoolsContext.Provider value={basicPools}>
+      {children}
+    </BasicPoolsContext.Provider>
+  )
+}
+
+/**
+ * Get all info about pools that can be read on chain.
+ * Excludes price data and user data.
+ * Will be replaced with registry.
+ */
+export async function getPoolsBaseData(
+  library: Web3Provider,
+  chainId: ChainId,
+): Promise<SwapInfo[]> {
+  const targetPools = Object.keys(POOLS_MAP).filter(
+    (poolName) => !!POOLS_MAP[poolName].addresses[chainId],
+  ) as PoolName[]
+  const poolsData = await Promise.all(
+    targetPools.map((poolName) => getSwapInfo(library, chainId, poolName)),
+  )
+  return poolsData.filter(Boolean) as SwapInfo[]
+}
+
+export async function getSwapInfo(
+  library: Web3Provider,
+  chainId: ChainId,
+  poolName: PoolName,
+): Promise<SwapInfo | null> {
+  try {
+    // Constants
+    const pool = POOLS_MAP[poolName]
+    const _metaSwapAddress = pool.metaSwapAddresses?.[chainId]?.toLowerCase()
+    const _internalAddress = pool.addresses[chainId]?.toLowerCase() // @dev currently points to metaswapDeposit
```

Comment:
Why are these variables underlined?

Reply:
I was trying to denote that these variables shouldn't be used by themselves (eg directly returned like some of the other vars). I'm open to a clearer pattern for denoting this if you have one. I'm kind of borrowing from python's `_` to represent ignored variables as in `const [_, b] = getPair()`